azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ContainerServiceImpl.java

```diff
@@ -213,12 +227,16 @@ private ContainerServiceImpl withDiagnosticsProfile(boolean enabled) {
     }
 
     @Override
-    public Update withAgentCount(int agentCount) {
+    public Update withAgentVMCount(int agentCount) {
         if (agentCount < 0 || agentCount > 100) {
             throw new RuntimeException("Agent pool count  must be in the range of 1 to 100 (inclusive)");

```

Comment:
we shouldn't be throwing this exception - it;'s for the service to impose these limits. (its business logic may change over time)